### DISCUSSION TITLE
MagickWand Method

### DISCUSSION THREAD

**Discussion by programmerarunkumar** [ImageMagick/ImageMagick#2462] [Help]
convert -debug "all" Image7.jpeg -resize 1600x1600! Image07.jpeg

What are the magick wand method  in C , need to use to convert the image with debug ?

**Comment by snibgo** [comment]
Umm... See your other thread https://github.com/ImageMagick/ImageMagick/discussions/2427

**Comment by programmerarunkumar** [comment]
> convert -debug "all" Image7.jpeg -resize 1600x1600! Image07.jpeg
> 
> What are the magick wand method in C , need to use to convert the image with debug ?

What is the method for debug in MagickWand

**Comment by snibgo** [comment]
As I said in that thread, call SetLogEventMask(). If you look at the source code of IM programs, such as convert.c, this is how the "-debug" option is implemented.

**Comment by programmerarunkumar** [comment]
I need the ImageMagick logs by changing the configuration of the ImageMagick

Any such way to get the ImageMagick logs ?

**Comment by snibgo** [comment]
I don't understand the question.

The command line option "-debug XX" can be implemented by calling SetLogEventMask("XX") where XX is any valid debug option, or a comma-separated list of options.

Is this "the ImageMagick logs" you mean? If not, please show a sample of the logs you mean, and the command that creates them.

**Reply by programmerarunkumar** [reply]
>The command line option "-debug XX" can be implemented by calling SetLogEventMask("XX") where XX is any valid debug option, or a comma-separated list of options.

Is there any other way to get logs ?

**Comment by emcconville** [comment]
The Magick++'s library has the `Magick::Options::debug()` method, but the MagickWand library does not. 

> Any such way to get the ImageMagick logs ?

Set the `MAGICK_DEBUG` environment variable. 

```console
$ export MAGICK_DEBUG=All
$ ./myMagickWandProgram
```
As @snibgo pointed out in #2427, you can do this within the program with `SetLogEventMask()`

```c
#include <MagickCore/MagickCore.h>

/* ... */
MagickWandGenesis();
(void) SetLogEventMask("All");
```

MagickCore is a dependent of MagickWand, so you should have no issues including + compiling.

**Comment by snibgo** [comment]
As @emcconville says. However, provided we have ...
```
#include <MagickWand/MagickWand.h>
```
... we don't also need ...
```
#include <MagickCore/MagickCore.h>
```
